packages/block-library/src/embed/util.js

```diff
@@ -311,6 +326,14 @@ export const getMergedAttributesWithPreview = (
 	ignorePreviousClassName = false
 ) => {
 	const { allowResponsive, className } = currentAttributes;
+
+	// Aspect ratio classes are removed when the embed URL is updated.
+	// If the embed already has an aspect ratio class, that means the URL has not changed.
+	// Which also means no need to regenerate it.
+	if ( hasAspectRatioClass( className ) ) {
+		return currentAttributes;
+	}
```

Comment:
@TylerB24890 I'm wondering if we could rely upon determining if the attributes should be updated from the preview data just by the presence of aspect ratio classnames 🤔 . From the testing I've done so far, looks like the attributes remain the same, but I'm not sure if this might lead to side effects in the future.

I'm thinking that this check might be done in `getAttributesFromPreview` function instead, specifically in the part we're getting the classnames, so we would only modify the `classname` attribute with this logic:
https://github.com/WordPress/gutenberg/blob/5695f47ac1bf5c4df5bef1338b4e25cb3772a3a6/packages/block-library/src/embed/util.js#L301-L305

So, in case the block already has aspect ratio classes, we would keep them instead of calculating new ones, wdyt?

Following this approach, we would need also to stop setting the `ignorePreviousClassName` argument to `true` when receiving the preview, so we don't ignore the current class names:
https://github.com/WordPress/gutenberg/blob/5695f47ac1bf5c4df5bef1338b4e25cb3772a3a6/packages/block-library/src/embed/edit.js#L147-L151

In fact, I introduced the `ignorePreviousClassName` parameter for addressing [this issue](https://github.com/wordpress-mobile/gutenberg-mobile/issues/4059) in https://github.com/WordPress/gutenberg/pull/35326. However, I checked that with the changes you made in relation to removing the aspect ratio classes when the embed URL changes, we no longer require it.

Reply:
No worries at all @TylerB24890 , thank you very much for making the updates 🙇. Sorry I couldn't review this earlier, I'm planning to take a look today 👀 .